src/geckomat/gather_kcats/selectKcatValue.m

```diff
@@ -55,18 +55,28 @@
 % Make vector with single kcat value per reaction
 idxInModelUnique = unique(idxInModel);
 selectedKcats    = zeros(numel(idxInModelUnique),1);
+selectedSource   = cell(numel(selectedKcats),1);
+if ~isfield(kcatList,'kcatSource')
+    kcatList.kcatSource = cell(numel(selectedKcats),1);
```

Comment:
`kcatList.kcatSource` should be same size as `kcats` field in `kcatList`. If this is based on` selectedKcats` will get since `selectedKcats` can contain less elements than `kcatList`. 

```suggestion
    kcatList.kcatSource = cell(numel(selectedKcats),1);
```